waspls/src/Wasp/LSP/Server.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ExplicitNamespaces #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Wasp.LSP.Server
+  ( run,
+  )
+where
+
+import qualified Control.Concurrent.MVar as MVar
+import Control.Monad.IO.Class (MonadIO (liftIO))
+import qualified Control.Monad.Trans.Except as Except
+import qualified Control.Monad.Trans.State.Strict as State
+import qualified Data.Aeson as Aeson
+import Data.Default (Default (def))
+import qualified Data.Text as Text
+import qualified Language.LSP.Server as LSP
+import qualified Language.LSP.Types as J
+import System.Exit (ExitCode (ExitFailure), exitWith)
+import qualified System.Log.Logger
+import Wasp.LSP.Core (ServerConfig, ServerM, Severity (..), State)
+import Wasp.LSP.Handlers
+
+run :: Maybe FilePath -> IO ()
```

Comment:
We could rename this one `serve` maybe? To indicate it runs until stopped?

Reply:
Done